src/ansys/fluent/core/fluent_connection.py

```diff
@@ -257,77 +217,15 @@ def __init__(
 
         self._remote_instance = remote_instance
         self.launcher_args = launcher_args
-        self._finalizer = weakref.finalize(
-            self,
-            FluentConnection._exit,
-            self._channel,
-            self._cleanup_on_exit,
-            self.scheme_eval,
-            self.datamodel_service_se,
-            self.datamodel_events,
-            self.transcript,
-            self.events_manager,
-            self.monitors_manager,
-            self._remote_instance,
-        )
```

Comment:
Shouldn't we keep `_finalizer` and the `exit()` function in FluentConnection? They are closing the "Fluent connection" after all. 

I believe this is also what @seanpearsonuk meant when he said "exit belongs here" (in FluentConnection), assuming I didn't misunderstand it.

Reply:
I am still re-structuring it. I'll update once the final structure is ready.

Reply:
Sorry, I thought it was ready for review. I converted the PR back to draft (the button was below the reviewers panel).

Reply:
Thank you @raph-luc. No issues, I had only made it ready for review, sorry for the confusion :).